File: C++/Longest Common Subsequence/shortest_common_supersequence.cpp
#include<bits/stdc++.h>
using namespace std;

int lcs(string a, string b){
    int m = a.length();
    int n = b.length();
    int dp[n+1][m+1];

    for(int i=0; i<=m; i++){
        for(int j=0; j<=n; j++){
            if(i==0 || j==0){
                dp[i][j] = 0;
            } else if(a[i-1]==b[j-1]){
                dp[i][j] = 1 + dp[i-1][j-1];
            } else{
                dp[i][j] = max(dp[i-1][j], dp[i][j-1]);
            }
        }
    }
    return dp[m][n];
}

int superseq(string a, string b){
    int m = a.length();
    int n = b.length();
    return m + n - lcs(a, b);
}

int main(){
    int test;
    string a, b;
    cin>>test;
    while(test--){
        cin>>a;
        cin>>b;
        cout<<superseq(a, b)<<'\n';
    }
}

File: Python/Longest Common Subsequence/shortest_common_supersequence.py
def lcs(a, b):
    m = len(a)
    n = len(b)
    dp = [[None for _ in range(n+1)] for _ in range(m+1)]

    for i in range(m+1):
        for j in range(n+1):
            if(i==0 or j==0):
                dp[i][j] = 0
            elif(a[i-1] == b[j-1]):
                dp[i][j] = dp[i-1][j-1] + 1
            else:
                dp[i][j] = max(dp[i-1][j], dp[i][j-1])
    return dp[m][n]

def superseq(a, b):
    m = len(a)
    n = len(b)
    return m + n - lcs(a, b)

test = int(input())
for i in range(test):
    a = input()
    b = input()
    print(superseq(a, b))